File: LeetCode/Medium/0120-triangle/0120-triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        for (int i = triangle.size() - 2; i > -1; i--)
            for (int j = 0; j < i+1; ++j) 
                triangle[i][j] = min(
                    triangle[i][j] + triangle[i+1][j], 
                    triangle[i][j] + triangle[i+1][j+1]);
        
        return triangle[0][0];
    }
};

File: LeetCode/Medium/0120-triangle/0120-triangle.go
func minimumTotal(triangle [][]int) int {
    for i := len(triangle) - 2; i > -1; i-- {
        for j := 0; j < i + 1; j++ {
            triangle[i][j] = min(triangle[i][j] + triangle[i+1][j], triangle[i][j] + triangle[i+1][j+1])
        }
    }
    
    return triangle[0][0]
}

func min(a, b int) int {
    if a < b { return a }
    return b
}